Frameworks/CoreGraphics/CGPath.mm

```diff
@@ -127,6 +145,23 @@ CFTypeID CGPathGetTypeID() {
 static Boolean __CGPathEqual(CFTypeRef cf1, CFTypeRef cf2) {
     __CGPath* path1 = (__CGPath*)cf1;
     __CGPath* path2 = (__CGPath*)cf2;
+
+    path1->closePath();
```

Comment:
We must close the path here. Any D2D1 operations require closed figures otherwise an error state is reached. This closes the path strictly for D2D1 operations and any future operations will re-open the path as necessary.

---
In reply to: [85667239](https://github.com/Microsoft/WinObjC/pull/1255#discussion_r85667239) [](ancestors = 85667239)